tasks/main.yml

```diff
@@ -34,6 +34,31 @@
   import_tasks: ca.yml
   when: ca_ca_host == inventory_hostname
 
+- name: Set temporary variable ca_san to hostname, FQDN, and inventory hostname
+  set_fact:
+    ca_san: >-
+      DNS:{{ ansible_hostname }},
+      DNS:{{ ansible_fqdn }},
+      DNS:{{ inventory_hostname }}
+
+- name: Add IP addresses and loopback address for etcd >= 3.2.0 to ca_san
+  set_fact:
+    ca_san: >-
+      {{ ca_san }},
+      IP:127.0.0.1,
+      {%- for host in vars['play_hosts'] -%}
+        IP:{{ hostvars[inventory_hostname]['ansible_default_ipv4']['address'] }}
+        {%- if not loop.last -%}
```

Comment:
Yes, I can see these three options:
- Remove the task "Add IP addresses and loopback address for etcd >= 3.2.0 to ca_san" and the default variable `ca_etcd` to just keep ca_san_custom. So the user can specify which IP addresses etc. have to be added.
- Loop over a group. Requires an additional default variable.
- Add a disclaimer to the Readme.md

Reply:
Hm... I think the last two combined sound like a way to go. My suggestion:
* Introduce a new variable:  ca_etcd_group . Hosts that are members in this group are counted as part of an etcd setup and will be included in the SAN of certificates on etcd hosts. Maybe this could even replace activating etcd as a whole. When it's set, etcd certificates are created if not, then not
* Add a paragraph to Readme.md explaining how to use the role when you have etcd

The overall idea is to run the role once over all hosts in the inventory. When it's done running, there should be the following:
* CA on the ca host
* Every host (including CA, etcd, Logstash etc) has their own key and certificate
* If server certificates are activated: Every host (including CA, etcd, Logstash etc) has their own key and certificate
* If Logstash keys are enabled, Every host (including CA, etcd, Logstash etc) has their own key in pkcs8 certificate
* If etcd is enabled, all etcd hosts have a key and certificate suitable for etcd additional to the others

This way we won't have collisions between tools.

Reply:
Changes:
- Added tasks to create certificates additionally in seperate runs instead of modifying the standard certificate creation
- Established a new variable ca_etcd_hosts to define IP addresses that have to be added to the SAN for etcd to work with TLS
- Established a variable with set_fact to save the SAN in a variable and to manipulate it (e.g removing spaces with regex_replace)